docs: getting started guide use pipe before introduction

ajitsinghkaler opened the pull request:
in getting started guide pipes are not intoduced anywhere but are used in guide added refrence to pipes for better consistency in the tutorial

Fixes #36375

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Pipes used before it were referenced

Issue Number: #36375


## What is the new behavior?
Pipes referenced on the first tutorial when they are used

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

mary-poppins commented:
You can preview 18d35ef at https://pr36584-18d35ef.ngbuilds.io/.

mary-poppins commented:
You can preview 18d35ef at https://pr36584-18d35ef.ngbuilds.io/.

gkalpak commented on aio/content/start/start-data.md:
```diff
@@ -101,6 +101,12 @@ This section walks you through using the cart service to add a product to the ca
+    
+    <div class="alert is-helpful">
+
+    The operator `|` used for currency on product.price is called a pipe. For more information about Angular pipes, see [Pipes](guide/pipes "Pipes").
```
`product.price` --> `` `product.price` ``

Also, the sentence now makes it sound as if `|` is called a pipe (which it's not). We should re-phrase to make it clear that `|` is the "pipe operator" and `currency` is the pipe (which is not a great name, but that's what we gone with 🤷‍♂️).

aikidave replied:
Thanks for this change! Here's a suggestion that I think might help. It puts more focus on the currency pipe than the operator, and provides a brief description of what a pipe is for.

```suggestion
    The line, `<h4>{{ product.price | currency }}</h4>` uses the `currency` pipe to transform `product.price` from number to a currency string. A pipe is a way you can transform data in your HTML template. For more information about Angular pipes, see [Pipes](guide/pipes "Pipes").
```

gkalpak reviewed (commented):
Thx, @ajitsinghkaler!

Please use proper sentences in the commit message to make it more readable (i.e. use capitalization, punctuation, etc.):

> In "Getting started" guide pipes are not intoduced anywhere but are used in the guide.
> Added refrence to pipes for better consistency in the tutorial.

aikithoughts reviewed (commented):
Thanks for pointing this out! I offered a suggestion. Let me know what you think.

ajitsinghkaler committed:
docs: getting started guide use pipe before introduction

In "Getting started" guide pipes are not intoduced anywhere but are used in the guide.
Added refrence to pipes for better consistency in the tutorial.

Fixes #36375

ajitsinghkaler commented:
@aikidave @gkalpak made the changes please have a look

mary-poppins commented:
You can preview c8252e9 at https://pr36584-c8252e9.ngbuilds.io/.

aikidave commented on aio/content/start/start-data.md:
```diff
@@ -101,6 +101,12 @@ This section walks you through using the cart service to add a product to the ca
+    
+    <div class="alert is-helpful">
+
+    The line, `<h4>{{ product.price | currency }}</h4>` uses the `currency` pipe to transform `product.price` from number to a currency string. A pipe is a way you can transform data in your HTML template. For more information about Angular pipes, see [Pipes](guide/pipes "Pipes").
```
```suggestion
    The line, `<h4>{{ product.price | currency }}</h4>` uses the `currency` pipe to transform `product.price` from a number to a currency string. A pipe is a way you can transform data in your HTML template. For more information about Angular pipes, see [Pipes](guide/pipes "Pipes").
```

aikidave replied:
Missing an "a". Otherwise this looks good!

ajitsinghkaler committed:
fixup! docs: getting started guide use pipe before introduction

Co-Authored-By: David Shevitz <dshevitz@google.com>

mary-poppins commented:
You can preview e678339 at https://pr36584-e678339.ngbuilds.io/.

ajitsinghkaler commented:
@aikidave @gkalpak made the change please have a look

ajitsinghkaler commented:
@akidave can you please add the merge label
